### PR TITLE
[Feature #160215754] Add a get all questions endpoint with tests

### PR DESCRIPTION
**Pull request by maxwellgithinji**
**What is this PR supposed to do?**
Create a User endpoint to `GET` all posted questions including tests

**Descriptions of tasks to be completed**
* `GET` a request from the user to display all questions

**How can this be manually tested?**
* Clone this repo and install all dependencies
* Run app 
* open postman and enter `/api/v1/questions` using a  `GET` method

**What are the related PT stories?**
https://www.pivotaltracker.com/story/show/160215754

### CODE REVIEW

File: API/helper/helper.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+"""
+JSON Helper module
+"""
+import json
+
+
+def post_json(client, url, json_dict):
+    """gets json dict, url and client for the post method"""
```
Comment (Dhul-Husni) [inline API/helper/helper.py]: Please be explicit on params, and return in docs

File: API/helper/helper.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
+"""
+JSON Helper module
+"""
+import json
+
+
+def post_json(client, url, json_dict):
```
Comment (Dhul-Husni) [inline API/helper/helper.py]: Nice

File: API/helper/helper.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    return client.post(
+        url,
+        data=json.dumps(json_dict),
+        content_type='application/json'
+        )
+
+
+def json_of_response(response):
+    """Decode JSON from response"""
+    return json.loads(response.data.decode('utf8'))
```
Comment (Dhul-Husni) [inline API/helper/helper.py]: Nice :)

File: API/models/models.py [hunk 4]
```diff
@@ -0,0 +1,11 @@
+"""
+App models using dictionaries ans list datastructures
```
Comment (Dhul-Husni) [inline API/models/models.py]: nit: small spelling error, should be `and`

File: API/models/models.py [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+"""
+
+ST_MODEL = {}
+
+ST_MODEL['questions'] = []
+ST_MODEL['answers'] = []
+
+QUESTIONS = ST_MODEL['questions'] = []
+ANSWERS = ST_MODEL['answers'] = []
```
Comment (Dhul-Husni) [inline API/models/models.py]: why do we have this
```
ST_MODEL['questions'] = []
ST_MODEL['answers'] = []
``` 
and still set this
```
QUESTIONS = ST_MODEL['questions'] = []
ANSWERS = ST_MODEL['answers'] = []
```

File: API/run.py [hunk 6]
```diff
@@ -0,0 +1,12 @@
+"""
+app runner module
```
Comment (Dhul-Husni) [inline API/run.py]: Doesn't make sense :)

File: API/tests/test_config.py [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+class TestDevelopmentConfig(TestCase):
+    """Configs test class"""
+    def create_app(self):
+        """app initializer for testing config"""
+        app = create_app('testing')
+        return app
+
+
+    def test_development_config(self):
```
Comment (Dhul-Husni) [inline API/tests/test_config.py]: Nice :)

File: API/tests/test_config.py [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+        self.assertEqual(app.config['SECRET_KEY'], 'What is your secret secret?')
+
+
+    def test_production_config(self):
+        """Production config test module"""
+        app = create_app('production')
+        self.assertTrue(app.config['DEBUG'] is False)
+        self.assertEqual(app.config['TESTING'], False)
+        self.assertEqual(app.config['SECRET_KEY'], 'What is your secret secret?')
+
```
Comment (Dhul-Husni) [inline API/tests/test_config.py]: Add new line

File: API/tests/test_get_questions.py [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+class TestGetQuestions(BaseConfig):
+    """Questions related endpoints test class"""
+    def test_get_all_questions(self):
+        """Get all questions test module"""
+        with self.client:
+            self.get_questions = self.client.get(
+                '/api/v1/questions'
+            )
+            self.assertEqual(self.get_questions.status_code, 200)
+            self.question_details = json_of_response(self.get_questions)
```
Comment (Dhul-Husni) [inline API/tests/test_get_questions.py]: Nice :)

**Comment by maxwellgithinji** [comment]
Thanks for the review. I will work on the feedback ASAP.

File: API/tests/test_get_questions.py [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+            self.assertEqual(self.question_details['message'], "No questions yet")           
+            #Implemet this after doing the post question endpoint
```
[the remark this answers is not in the document]
Comment (maxwellgithinji) [reply API/tests/test_get_questions.py]: noted

File: API/app/__init__.py [hunk 11]
```diff
@@ -9,8 +9,14 @@
 from app.config import APP_CONFIG
 
 
-def create_app(config_name=None):
-    """A helper function to create app instance"""
+def create_app(config_name='develoment'):
```
Comment (graycadeau) [inline API/app/__init__.py]: Slight spelling error there :-)  development

File: API/helper/helper.py (hunk 2 above)
Comment (graycadeau) [reply API/helper/helper.py]: :+1:

File: API/app/__init__.py (hunk 11 above)
Comment (maxwellgithinji) [reply API/app/__init__.py]: noted

File: API/app/__init__.py [hunk 12]
```diff
@@ -9,8 +9,14 @@
...
+    """
+    create app `function`
+    
+	param: str config_name: `str` to get dict config object
+    :type config_name: `dict key` in `APP_CONFIG`
+	return: app
+    """
```
[the remark this answers is not in the document]
Comment (Dhul-Husni) [reply API/app/__init__.py]: @maxwellgithinji Please look into this

[the remark this answers is not in the document]
Comment (Dhul-Husni) [reply API/app/__init__.py]: the doc string should be more like 👆 this
its simple and explains everything.  saying `str to get dict config object` isn't as explicit as saying `app environment`
also you are repeating yourself saying `str` twice

File: API/run.py [hunk 13]
```diff
@@ -1,8 +1,8 @@
 """
 This module creates an instace of the application;
-`APP = create_app(config_name)`By passing the
+`APP = create_app(config_name)` By passing the
 current configuration in the environment; `os.getenv(`FLASK_CONFIG`)`
-then the configuration exists, it runs the app `APP.run()`
+then if the configuration exists, it runs the app `APP.run()`
 """
```
Comment (Dhul-Husni) [inline API/run.py]: This should be 
```python
"""
run the app server when executed directly
"""
```
thats all, `Keep it simple stupid`

File: API/tests/base.py [hunk 14]
```diff
@@ -8,18 +8,27 @@
 
 
 class BaseConfig(TestCase):
-    """Base class for app testcases"""
+    """
+    Base class for app testcases
+
+    :param: `class` TestCase: the parent class
+    """
```
Comment (Dhul-Husni) [inline API/tests/base.py]: say 
`Base test class`

File: API/tests/test_config.py [hunk 15]
```diff
@@ -8,9 +8,18 @@
 
 
 class TestDevelopmentConfig(TestCase):
-    """Configs test class"""
+    """
+    Configs test class
+
+    param: `class` Testcase: The parent class
+    """
```
Comment (Dhul-Husni) [inline API/tests/test_config.py]: here too

File: API/app/__init__.py [hunk 16]
```diff
@@ -9,8 +9,14 @@
 from app.config import APP_CONFIG
 
 
-def create_app(config_name=None):
-    """A helper function to create app instance"""
+def create_app(config_name='develoment'):
+    """
+    create app `function`
```
[the remark this answers is not in the document]
Comment (Dhul-Husni) [reply API/app/__init__.py]: please see this doc
you dont need to explicitly state this is a function, it is fairly obvious :)

**Comment by maxwellgithinji** [comment]
👍

**Pull request merged**